### PR TITLE
Add cooldown config var

### PR DESCRIPTION
**Pull request by papiersnipper**
Implements issue #298.

### CODE REVIEW

File: bot.py [hunk 1]
```diff
@@ -574,6 +577,18 @@ def main_color(self) -> int:
                 )
                 guild_age = self.config.remove("guild_age")
 
+        if not isinstance(thread_cooldown, isodate.Duration):
+            try:
+                thread_cooldown = isodate.parse_duration(thread_cooldown)
+            except isodate.ISO8601Error:
+                logger.warning(
+                    "The account age limit needs to be a "
```
Comment (Taaku18) [inline bot.py]: "The account age limit", copy-paste problem.

File: bot.py [hunk 2]
```diff
@@ -591,6 +606,15 @@ def main_color(self) -> int:
             logger.warning("Error with 'guild_age'.", exc_info=True)
             self.config.remove("guild_age")
 
+        try:
+            last_log = (await self.api.get_user_logs(message.author.id))[-1]
+            cooldown = datetime.strptime(last_log["created_at"].split(".")[0], '%Y-%m-%d %H:%M:%S') + isodate.parse_duration(self.config["thread_cooldown"])
```
Comment (Taaku18) [inline bot.py]: `isodate.parse_duration(self.config["thread_cooldown"])` should be replaced with `thread_cooldown`. and parse created_at with `https://docs.python.org/3/library/datetime.html#datetime.datetime.fromisoformat`.

File: bot.py [hunk 3]
```diff
@@ -591,6 +606,15 @@ def main_color(self) -> int:
...
+        except IndexError:
```
Comment (Taaku18) [inline bot.py]: Wouldn't that cause name error later on?

File: bot.py [hunk 4]
```diff
@@ -663,6 +687,16 @@ def main_color(self) -> int:
                         self.blocked_users.pop(str(message.author.id))
                 else:
                     logger.debug("User blocked, user %s.", message.author.name)
+        elif cooldown > now:
+            delta = human_timedelta(cooldown)
```
Comment (Taaku18) [inline bot.py]: Don't believe `reaction` is defined.

File: core/config.py [hunk 5]
```diff
@@ -5,11 +5,11 @@
 import typing
 from copy import deepcopy
 
-from dotenv import load_dotenv
 import isodate
 
 import discord
 from discord.ext.commands import BadArgument
+from dotenv import load_dotenv
```
Comment (Taaku18) [inline core/config.py]: `dotenv` should be back where it was, pip8 import grouping.

File: core/config.py [hunk 6]
```diff
@@ -153,7 +154,7 @@ def populate_cache(self) -> dict:
         config_help_json = os.path.join(
             os.path.dirname(os.path.abspath(__file__)), "config_help.json"
         )
-        with open(config_help_json, "r", encoding="utf-8") as f:
+        with open(config_help_json, "r", encoding="utf-8-sig") as f:
```
Comment (Taaku18) [inline core/config.py]: Why? https://stackoverflow.com/a/2223926/6622817.

File: bot.py (hunk 1 above)
Comment (papiersnipper) [reply bot.py]: Exposed 👀

File: bot.py (hunk 4 above)
Comment (Taaku18) [reply bot.py]: And also, why don't you implement it the same way as account_age and guild_age, although they look a bit messy. The reason for how the others are implemented is to log it in the block reason and only show the "Message not sent" message once in a user's DM. If you want, just leave it and I can convert it later on.

**Comment by Taaku18** [comment]
It seems like because of the pull app enabled on your fork, your changes were overwritten.

**Pull request merged**